### PR TITLE
Mips initial support

### PR DESCRIPTION
**Pull request by wargio**
https://github.com/wargio/r2dec-js/commit/90c0af42f1515bb806a24d8bfbac99d81bb2291e

### CODE REVIEW

File: mips/cond.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+            //l[start].offset = e;
+        } else {
+            //this should never happen, but let's add it anyway..
+            l[start] = e;
+        }
+        */
+        return l;
+    };
+
+    var branch = {
```
Comment (radare) [inline mips/cond.js]: what about autogenerating this object from an array of {'beqz', 'NE', ...} and make this generic to simplify the code of all the archs

File: mips/return.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+module.exports = (function() {
+    var to_asm = function(e) {
+        var j;
+        var asm = e[0] + " ";
+        for (j = 1; j < e.length - 1; ++j) {
```
Comment (radare) [inline mips/return.js]: just put the asm = e.join(', ')

File: mips/return.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        if (j < e.length)
+            asm += e[j];
+        return asm;
+    };
+
+    var mem = {
+        'b': function(l, start) {
+            var i;
+            var offset = 'label_' + l[start].opcode[1].toString(16);
+            for (var i = 0; i < l.length; i++) {
```
Comment (radare) [inline mips/return.js]: This can be simplified with filter and map

File: mips/cond.js (hunk 1 above)
Comment (wargio) [inline mips/cond.js]: Good idea!

File: mips/return.js (hunk 2 above)
Comment (wargio) [inline mips/return.js]: Actually that's old code reuse.. That needs to be changed everywhere.

**Pull request merged**